Havit.Blazor.Components.Web.Bootstrap.Documentation/Pages/Components/Concepts/SettingsAndDefaults_Documentation.razor

```diff
@@ -66,4 +48,42 @@ HxButton.Defaults.Icon = BootstrapIcon.Fullscreen;
 		HxOffcanvas.Defaults.BackdropEnabled = false;
 	}
 }
-</CodeSnippet>
\ No newline at end of file
+</CodeSnippet>
+
+<SectionTitle Id="overriding" Title="Overriding" />
+<p>When a component decides what value to adhere to it heeds this flow: <strong>parameter > settings > defaults</strong> meaning that when the parameter property is <code>null</code>, then the value is taken from settings, if that is also <code>null</code>, then the value is taken from application-wide defaults. In case all of the above are <code>null</code>, an exception is thrown since this is an unsupported scenario.</p>
+<Demo Type="typeof(SettingsAndDefaults_Demo_Overriding)" />
+
+<h2 id="settings">Settings</h2>
+<p>Component settings are a way of specifying component parameters. They let you bundle parameters together. That way, component settings can be easily sent across the application and applied in multiple places without the need to send the values individually. This comes in handy for instance when storing user preferences.</p>
+
+<SectionTitle Id="settings-demo">Demo</SectionTitle>
+<p>Applying component settings is as easy as instantiating a settings object for the target component and assigning it to the <code>Settings</code> parameter as shown below.</p>
+<Demo Type="typeof(SettingsAndDefaults_Demo_Settings)" />
+
+<p>Setting application-wide collection of settings used for varying buttons serving different purposes.</p>
+<CodeSnippet Language="cs">public static class AppButtonSettings
+{
+	/// <summary>
+	/// We can derive the settings from the defaults using the with statement.
+	/// </summary>
+	public static ButtonSettings MainButton { get; } = HxButton.Defaults with
+	{
+		Color = ThemeColor.Primary,
+		Icon = BootstrapIcon.Fullscreen
+	};
+
+	public static ButtonSettings SecondaryButton { get; } = new()
+	{
+		Color = ThemeColor.Secondary,
+		Outline = true
+	};
+
+	public static ButtonSettings CloseButton { get; } = new()
+	{
+		Color = ThemeColor.Danger,
+		Outline = true,
+		Icon = BootstrapIcon.X
+	};
+}
+</CodeSnippet>
```

Comment:
Extract the code to `AppButtonSettings.codesnippet.cs` for better maintainability.